### PR TITLE
Updated create-baremetals and some clean-ups

### PR DESCRIPTION
**Pull request by zhanggbj**
1. Added more frequently-used server spec for create-baremetals
2. Fixed minor issues when to create a baremetal pool
3. Some clean-ups

[#121655475](https://www.pivotaltracker.com/story/show/121655475)

And code for commit a087ec5 and 3c0e5d7 have been merged in another PR.

### CODE REVIEW

**Comment by zhanggbj** [comment]
Hi dr.max @maximilien ,

This is ready for review. Thanks.

File: cmds/bmp/update_state_command.go [hunk 1]
```diff
@@ -101,20 +101,25 @@ func (cmd updateStateCommand) Execute(args []string) (int, error) {
 // Private Methods
 
 func (cmd updateStateCommand) isValidState(state string) bool {
-	validState := map[string]bool{
-		"bm.state.new":     true,
-		"bm.state.using":   true,
-		"bm.state.loading": true,
-		"bm.state.failed":  true,
-		"bm.state.deleted": true,
+	switch state {
```
Comment (maximilien) [inline cmds/bmp/update_state_command.go:104]: I think you can simplify this code by having many `case` statements and one `return true` statement?

**Comment by maximilien** [comment]
Ok looks good. One comment (look into it) but merging this. Cheers :+1:

File: cmds/bmp/update_state_command.go (hunk 1 above)
Comment (zhanggbj) [inline cmds/bmp/update_state_command.go:104]: Yes, right, I'll update it next time:)

**Pull request merged**